File: leetcode/643/643.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        n = len(nums)

        sums = sum(nums[:k])
        avg = -10000
        for l in range(n-k+1):
            avg = max(avg, sums / k)

            r = l + k
            if r < n:
                sums = sums - nums[l] + nums[r]
        return avg

File: leetcode/643/643.c
double findMaxAverage(int* nums, int numsSize, int k){
    int sum = 0;
    double res = -999999;
    for (int i=0; i<numsSize; ++i) {
        sum += nums[i];
        if (i >= k-1) {
            double avg = (double)(sum) / k;
            res = res > avg ? res : avg;
            sum -= nums[i-k+1];
        }
    }
    return res;
}
